resources/assets/js/components/WordCounter/WordCounter.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect } from "react";
+import ProgressRing from "../ProgressRing";
+
+/*
+  Question: Should this component have the option to add a textarea component (or input element)?
+  example:
+    prop: textAreaComponent: React.ReactElement
+    ...
+    return (
+      {textAreaComponent && textAreaComponent}
+    )
+*/
+
+interface Message {
+  max: number;
+  message: string;
+}
+
+interface WordCounterProps {
+  numOfWords: number;
+  maxWords: number;
+  minWords: number;
+  warnings: Message[];
+  wordLimit: number;
+}
+
+const WordCounter: React.FunctionComponent<WordCounterProps> = ({
+  numOfWords,
+  minWords,
+  maxWords,
+  warnings,
+}): React.ReactElement => {
+  const message = (): string => {
+    let index = 0;
+    const warning = warnings.find(
+      ({ max }, i): boolean => {
+        index = i;
+        return (
+          warnings[i + 1] &&
```

Comment:
If you assume warnings are ordered from smallest max to largest (which I think has to be true for this to work), do you need to compare to `warnings[i+1]` at all?